src/Microsoft.AspNet.OData.Shared/DeltaOfTStructuralType.cs

```diff
@@ -101,12 +132,29 @@ public override Type StructuredType
             }
         }
 
+        internal IDictionary<string, object> DeltaNestedResources
+        {
+            get {return _deltaNestedResources; }
+        }
+
         /// <inheritdoc/>
         public override Type ExpectedClrType
         {
             get { return typeof(TStructuralType); }
         }
 
+        /// <summary>
+        /// Gets the enum type of <see cref="EdmDeltaEntityKind"/>.
+        /// </summary>
+        public EdmDeltaEntityKind DeltaKind { get; set; }
```

Comment:
>set [](start = 51, length = 3)

should this be set-able?  What if I have a DeltaDeletedEntityObject and I set this to "Entry"?

If the only two delta kinds we support are entry and deletedEntry, then we should just make this read-only, set when the constructor is called.